### DISCUSSION TITLE
Odd behaviour when manually jogging

### DISCUSSION THREAD

**Discussion by Festivejelly** [kachurovskiy/nanoels#186] [Q&A] [answered]
I needed to increase my backlash value on my Z as it was out slightly. Ive set it to 3000 but since doing that when I try to jog the Z axis even when the steps are set to 1mm it jogs 2.

Any idea why this might be? If I tell it to move with Gcode it seems to act as I expect. Is this to do with the backlash compensatation?

**Comment by kachurovskiy** [comment, marked as answer]
@Festivejelly depending on your acceleration, initial speed there's a maximum time between button down and up events that results in move by 1 step. If you hold the button just a tiny bit longer, it will take 2 steps instead of 1.

Since we need to allow for deceleration and ensure smooth motion, we make the decision of whether to take the next step before the previous one is over https://github.com/kachurovskiy/nanoels/blob/main/h4/h4.ino#L1105

```
void waitForPendingPosNear0(Axis* a) {
  while (abs(a->pendingPos) > a->motorSteps / 3) {
    taskYIELD();
  }
}
```

If you change `3` to some over number e.g. `5` you might get more time to lift up the finger and only end up with 1 step - but this could reduce deceleration time, make the motion a little more jerky and result in carriage over-traveling a bit due to the backlash since lead screw is pushing the carriage but is unable to stop it until the backlash is taken out. Proper deceleration is important so that the carriage doesn't have the momentum to keep going on it's own.
